### PR TITLE
Task 7 complete

### PR DESCRIPTION
**Pull request by SaraGMatos**
- POST endpoint for /api/treasures/:treasure_id/comments
- Controller and model for POST /api/treasures/:treasure_id/comments
-Testing suite deals with:
         - Successful response for POST /api/treasures/:treasure_id/comments
         - Error handling for incomplete body, invalid id data type, inexistent id, and wrong body type.
- Updated README
- Updated endpoints.json

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -143,6 +166,70 @@ describe("/api/articles", () => {
...
+        .then(({ body }) => {
+          expect(body.message).toBe("Bad request.");
+        });
+    });
+
+    test("POST 400: Responds with an adequate status and error message when the data type of the object's property values are not correct", () => {
+      return request(app)
+        .post("/api/articles/2/comments")
+        .send({ username: 200, body: false })
+        .expect(400)
```
Comment (pippaamy) [inline __tests__/app.test.js:207]: these tests are great - what if the username was "Pippa" - what error would we get then?

File: models/nc_news.models.js [hunk 2]
```diff
@@ -34,18 +34,43 @@ function fetchArticleById(article_id) {
...
+      if (articles.length === 0) {
         return Promise.reject({ status: 404, message: "Article not found." });
       }
-      return rows;
+    });
+}
+
+function addCommentByArticleId(article_id, { username, body }) {
+  const author = username;
+  const votes = 0;
```
Comment (pippaamy) [inline models/nc_news.models.js:63]: you could have a promise.reject here that if there was no username or body it wouldnt go on

File: controllers/nc_news.controllers.js [hunk 3]
```diff
@@ -34,18 +40,40 @@ function getArticleById(req, res, next) {
...
       next(err);
     });
 }
 
+function postCommentByArticleId(req, res, next) {
+  const { article_id } = req.params;
+  const postedCommentInfo = req.body;
+
+  Promise.all([
+    checkIfArticleExists(article_id),
```
Comment (pippaamy) [inline controllers/nc_news.controllers.js:60]: Im not sure we need to check if the article id exists here as we would get a psql error if it didnt whilst trying to add a comment :)

**Review by pippaamy — COMMENTED** [review]
Nice work Sara, everything youre doing is working - if you want to simplify it have a look at my suggestions :)

File: __tests__/app.test.js (hunk 1 above)
Comment (SaraGMatos) [reply __tests__/app.test.js:207]: Would it be a 400? It would trigger an psql error.

File: controllers/nc_news.controllers.js (hunk 3 above)
Comment (SaraGMatos) [reply controllers/nc_news.controllers.js:60]: Done - thank you :)

**Pull request merged**